src/app/GitUI/UserControls/FileStatusDiffCalculator.cs

```diff
@@ -315,7 +315,9 @@ GitRevision GetActualRevisionForRevision(GitRevision revision)
             }
 
             IGitModule module = GetModule();
-            IReadOnlyList<GitItemStatus> statuses = module.GetGrepFilesStatus(selectedRev.ObjectId, _fileStatusDiffCalculatorInfo.GrepArguments, applyAppSettings: !_fileStatusDiffCalculatorInfo.FileTreeMode, cancellationToken);
+            IReadOnlyList<GitItemStatus> statuses = !selectedRev.IsArtificial && _fileStatusDiffCalculatorInfo.GrepArguments == @"-e ""^"""
+                ? module.GetTreeFiles(selectedRev.ObjectId, full: true)
+                : module.GetGrepFilesStatus(selectedRev.ObjectId, _fileStatusDiffCalculatorInfo.GrepArguments, applyAppSettings: !_fileStatusDiffCalculatorInfo.FileTreeMode, cancellationToken);
```

Comment:
Will cleanup a little: Remove the hardcoded setting of GrepArguments  to ^ for file tree (use "") and move that to GetTreeFiles() for articicial (replace later with ls-files?) (maybe cancellation token too, but ls-tree does not take more than 200ms also in Linux repo so not long running).

Reply:
Thank you, and much simpler than feared. 🎉